src/openms/source/CHEMISTRY/ISOTOPEDISTRIBUTION/IsoSpec.cpp

```diff
@@ -61,41 +62,30 @@ namespace OpenMS
   {
   }
 
-  const std::vector<double>& IsoSpec::getMasses() {return masses_;}
-  const std::vector<double>& IsoSpec::getProbabilities() {return probabilities_;}
-
-  void IsoSpec::run_(Iso* iso)
+  std::vector<Peak1D> IsoSpec::run_(Iso& iso)
   {
     int tabSize = 1000;
     int hashSize = 1000;
 
-    IsoThresholdGenerator* generator = new IsoThresholdGenerator(std::move(*iso), threshold_, absolute_, tabSize, hashSize); 
-    Tabulator<IsoThresholdGenerator>* tabulator = new Tabulator<IsoThresholdGenerator>(generator, true, true, true, true); 
-
-    int size = tabulator->confs_no();
+    IsoThresholdGenerator generator(std::move(iso), threshold_, absolute_, tabSize, hashSize); 
 
-    masses_.clear();
-    masses_.reserve(size);
-    copy(&tabulator->masses()[0], &tabulator->masses()[size], back_inserter(masses_));
+    std::vector<Peak1D> distribution;
 
-    probabilities_.clear();
-    probabilities_.reserve(size);
-    copy(&tabulator->probs()[0], &tabulator->probs()[size], back_inserter(probabilities_));
+    while(generator.advanceToNextConfiguration())
+        distribution.emplace_back(Peak1D(generator.mass(), generator.eprob()));
```

Comment:
the only issues with this is that in OpenMS `Peak1D` stores the intensity in a float and not a double. So this means that we can only represent probabilities down to 1.17549e-038 with this interface. Maybe something to consider, especially in case somebody actually wanted access to the accurate numbers.